pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,323 @@
+		shouldBindPspt := true
+	*/
+
+	context.Core.ServiceAccounts("").AddHandler(
+		"universalServiceAccountHandler",
+		func(key string, obj *v1.ServiceAccount) error {
+			/*
+				r := func() {
+					mutex.Lock()
+					defer mutex.Unlock()
+					if shouldBindPspt {
+						shouldBindPspt = false
+
+						google11, err := m.clusterLister.Get("", "local")
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+						google11.Spec.DefaultPodSecurityPolicyTemplateName = "pspt1000"
+
+						_, err = m.clusters.Update(google11)
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+					}
+				}
+
+				if true {
+					r()
+				}
+			*/
+			logrus.Info("handle service account")
+
+			if obj == nil {
+				logrus.Infof("no service account provided, exiting")
+				// do nothing
+				return nil
+			}
+
+			namespaceName := obj.Namespace
+
+			logrus.Infof("svc acct name: %v namespace: %v", obj.Name, namespaceName)
+
+			// Check if a PSP exists for this cluster
+			policies, err := m.policyLister.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting policies: %v", err)
+			}
+
+			bindings, err := m.bindingLister.List(namespaceName, labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting role bindings: %v", err)
+			}
+
+			// get PSPT
+			// if no PSPT then get default
+			// if no default then exit
+			namespace, err := m.namespaceLister.Get("", namespaceName)
+
+			if err != nil {
+				return fmt.Errorf("error getting namespaces: %v", err)
+			}
+
+			annotation := namespace.Annotations[projectIdAnnotation]
+
+			if annotation == "" {
+				// no project is associated with namespace so don't do anything
+				logrus.Infof("no project is associated with namespace %v", namespaceName)
+				return nil
+			}
+
+			split := strings.Split(annotation, ":")
+
+			if len(split) != 2 {
+				return fmt.Errorf("could not parse project id annotation: %v", annotation)
+			}
+
+			clusterName, projectId := split[0], split[1]
+
+			projects, err := m.projectLister.List("", labels.Everything())
+
+			var project *v3.Project
+
+			for _, candidate := range projects {
+				if candidate.Name == projectId {
+					project = candidate
+					break
+				}
+			}
+
+			if err != nil {
+				return fmt.Errorf("error getting projects: %v", err)
+			}
+
+			podSecurityPolicyTemplateId := project.Spec.PodSecurityPolicyTemplateName
+
+			if podSecurityPolicyTemplateId == "" {
+				// check cluster
+				cluster, err := m.clusterLister.Get("", clusterName)
+
+				if err != nil {
+					return fmt.Errorf("error getting clusters: %v", err)
+				}
+
+				if cluster == nil {
+					return fmt.Errorf("could not find cluster %v", clusterName)
+				}
+
+				podSecurityPolicyTemplateId = cluster.Spec.DefaultPodSecurityPolicyTemplateName
+
+				if podSecurityPolicyTemplateId == "" {
+					logrus.Infof("No PSPTs found for project %v and cluster %v", projectId, clusterName)
```

Comment:
The most common scenario will be for there to be NO PSPT set. This would get very noisy in the logs. Either remove this or tune it down to debug